homeassistant/components/zone/__init__.py

```diff
@@ -135,7 +135,7 @@ def async_active_zone(
             is None
             # Skip zone that are outside the radius aka the
             # lat/long is outside the zone
-            or not (zone_dist - (radius := zone_attrs[ATTR_RADIUS]) < radius)
+            or not (zone_dist - (zone_radius := zone_attrs[ATTR_RADIUS]) < radius)
```

Comment:
This was a refactoring error. 😨  I added a test for non-zero radius since we didn't have one.